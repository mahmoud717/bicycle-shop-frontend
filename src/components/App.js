/* eslint-disable max-len */
/* eslint-disable no-console */
/* eslint-disable react/prop-types */
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { useEffect } from 'react';
import { connect } from 'react-redux';
import axios from 'axios';
import changeLoggedUser from '../redux/actions/auth_actions';
import About from './about';
import Navbar from './navbar';
import NotFound from './404';
import Home from './home';
import '../index.css';
import Login from '../containers/auth/login';
import Logout from '../containers/auth/logout';
import Signup from '../containers/auth/sign_up';
import BicycleList from '../containers/bicycles/bicycleList';
import Bicycle from '../containers/bicycles/Bicycle';
import NewOrder from '../containers/orders/NewOrder';
import Success from '../containers/orders/Success';
import User from '../containers/users/User';

function App({ authData, changeLoggedUser }) {
  useEffect(() => {
    axios.get('http://localhost:5000/api/v1/sessions/logged_in', { withCredentials: true })
      .then(response => {
        if (response.data) {
          changeLoggedUser(response.data.user, response.data.logged_in, response.data.orders);
        }
      })
      .catch(error => {
        console.error(error);
      });
  }, []);
  return (
    <BrowserRouter>
      <Navbar loggedIn={authData.loggedIn} userName={authData.user.name} userId={authData.user.id} />
      <div className="App m-0">
        <Switch>
          <Route exact path="/users/:id">
            <User authData={authData} />
          </Route>
          <Route exact path="/login">
            <Login authData={authData} />
          </Route>
          <Route exact path="/signup">
            <Signup authData={authData} />
          </Route>
          <Route exact path="/">
            <Home authData={authData} />
          </Route>
          <Route exact path="/bicycles">
            <BicycleList authData={authData} />
          </Route>
          <Route exact path="/logout">
            <Logout authData={authData} />
          </Route>
          <Route exact path="/about">
            <About authData={authData} />
          </Route>
          <Route exact path="/bicycles/:id">
            <Bicycle authData={authData} />
          </Route>
          <Route exact path="/orders/success">
            <Success authData={authData} />
          </Route>
          <Route exact path="/orders/new/:id">
            <NewOrder authData={authData} />
          </Route>
          <Route exact path="*">
            <NotFound authData={authData} />
          </Route>

        </Switch>

      </div>

    </BrowserRouter>
  );
}

const mapStateToProps = state => ({
  authData: state.auth,
});

const mapDispatchToProps = dispatch => ({
  changeLoggedUser:
  (user, loggedIn, userOrders) => dispatch(changeLoggedUser(user, loggedIn, userOrders)),
});
export default connect(mapStateToProps, mapDispatchToProps)(App);
