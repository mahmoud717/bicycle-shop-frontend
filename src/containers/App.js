/* eslint-disable max-len */
/* eslint-disable no-console */
/* eslint-disable react/prop-types */
import {
  BrowserRouter, Route, Switch,
} from 'react-router-dom';
import { useEffect } from 'react';
import { connect } from 'react-redux';
import axios from 'axios';

import changeLoggedUser from '../redux/actions/auth_actions';
import About from '../components/about';
import Navbar from '../components/navbar';
import NotFound from '../components/404';
import Home from '../components/home';
import '../index.css';
import Login from './auth/login';
import Logout from './auth/logout';
import Signup from './auth/sign_up';
import BicycleList from '../components/bicycles/bicycleList';
import Bicycle from '../components/bicycles/Bicycle';
import NewOrder from './orders/NewOrder';
import Success from '../components/orders/Success';
import User from './users/User';
import UserOrders from './users/UserOrders';
import CreateBicycle from './users/admin/CreateBicycle';
import AddOptions from './users/admin/addOptions';

function App({ authData, changeLoggedUser }) {
  useEffect(() => {
    axios.get('https://bicycle-shop-backend.herokuapp.com/api/v1/sessions/logged_in', {
      headers: {
        'Content-Type': 'application/json',
        Authorization: `bearer ${localStorage.getItem('token')}`,
      },
    })
      .then(response => {
        if (response.data) {
          changeLoggedUser(response.data.user, response.data.logged_in, response.data.orders, false);
          console.log(response.data);
        }
      });
    // .catch(error => {
    //   console.error(error);
    // });
  }, []);

  return (
    <BrowserRouter>
      <Navbar loggedIn={authData.loggedIn} userName={authData.user.name} userId={authData.user.id} />
      <div className="App m-0">
        <Switch>
          <Route exact path="/users/:id">
            <User authData={authData} />
          </Route>
          <Route exact path="/login">
            <Login authData={authData} />
          </Route>
          <Route exact path="/signup">
            <Signup authData={authData} />
          </Route>
          <Route exact path="/">
            <Home authData={authData} />
          </Route>
          <Route exact path="/bicycles">
            <BicycleList authData={authData} />
          </Route>
          <Route exact path="/logout">
            <Logout authData={authData} />
          </Route>
          <Route exact path="/about">
            <About authData={authData} />
          </Route>

          <Route exact path="/bicycles/create">
            <CreateBicycle authData={authData} />
          </Route>
          <Route exact path="/bicycles/:id/addOptions">
            <AddOptions authData={authData} />
          </Route>
          <Route exact path="/bicycles/:id">
            <Bicycle authData={authData} />
          </Route>
          <Route exact path="/orders/success">
            <Success authData={authData} />
          </Route>
          <Route exact path="/orders/new/:id">
            <NewOrder authData={authData} />
          </Route>
          <Route exact path="/users/:id/orders">
            <UserOrders authData={authData} />
          </Route>
          <Route exact path="*">
            <NotFound authData={authData} />
          </Route>

        </Switch>

      </div>

    </BrowserRouter>
  );
}

const mapStateToProps = state => ({
  authData: state.auth,
});

const mapDispatchToProps = dispatch => ({
  changeLoggedUser:
  (user, loggedIn, userOrders, loading) => dispatch(changeLoggedUser(user, loggedIn, userOrders, loading)),
});
export default connect(mapStateToProps, mapDispatchToProps)(App);
