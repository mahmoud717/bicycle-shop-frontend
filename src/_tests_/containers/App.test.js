import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import App from '../../containers/App';

const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <App />
  </Provider>,
);

describe('App Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });

  test('renders the page', () => {
    const page = component.find('.App');
    expect(page).toMatchSnapshot();
  });

  test('renders the login-access class', () => {
    const text = component.find('.login-access');
    expect(text).toMatchSnapshot();
  });

  test('renders the the route-trackings class', () => {
    const text = component.find('.route-trackings');
    expect(text).toMatchSnapshot();
  });
});
