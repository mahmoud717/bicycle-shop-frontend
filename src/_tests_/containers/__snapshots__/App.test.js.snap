// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders the login-access class 1`] = `ShallowWrapper {}`;

exports[`App Component renders the page 1`] = `ShallowWrapper {}`;

exports[`App Component renders the the route-trackings class 1`] = `ShallowWrapper {}`;
